### PR TITLE
Match Dependency Versions

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Slim the wepback entry point size by matching the peer dependency verisons. What is happening is hoist will attempt to flatten node_modules to the root level but isn't clever enough to flatten dependencies with mismatching minimum versions. This bloated the console output on install and caused our webpack entry point size to be ~6.81MiBs. Once these matched, the entry point size was ~3.07MiB.

<details><summary>Terminal Output</summary>
<p>

####  lerna bootstrap --hoist

```bash
> terra-framework@0.1.0 bootstrap:hoist /Users/er047227/terra-framework
> lerna bootstrap --hoist

lerna info version 2.11.0
lerna info versioning independent
lerna info Bootstrapping 26 packages
lerna info lifecycle preinstall
lerna WARN EHOIST_ROOT_VERSION The repository root depends on xfc@^1.2.1, which differs from the more common xfc@^1.0.0.
lerna WARN EHOIST_PKG_VERSION "terra-embedded-content-consumer" package depends on xfc@^1.0.0, which differs from the hoisted xfc@^1.2.1.
lerna WARN EHOIST_PKG_VERSION "terra-dialog-modal" package depends on terra-action-header@^1.2.0, which differs from the hoisted terra-action-header@^1.14.0.
lerna WARN EHOIST_PKG_VERSION "terra-modal-manager" package depends on terra-action-header@^1.27.0, which differs from the hoisted terra-action-header@^1.14.0.
lerna WARN EHOIST_PKG_VERSION "terra-navigation-side-menu" package depends on terra-action-header@^1.0.0, which differs from the hoisted terra-action-header@^1.14.0.
lerna WARN EHOIST_PKG_VERSION "terra-slide-panel-manager" package depends on terra-action-header@^1.26.0, which differs from the hoisted terra-action-header@^1.14.0.
lerna WARN EHOIST_PKG_VERSION "terra-aggregator" package depends on terra-arrange@^2.19.0, which differs from the hoisted terra-arrange@^2.3.0.
lerna WARN EHOIST_PKG_VERSION "terra-application-utility" package depends on terra-arrange@^2.1.0, which differs from the hoisted terra-arrange@^2.3.0.
lerna WARN EHOIST_PKG_VERSION "terra-application-layout" package depends on terra-button@^2.2.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-utility" package depends on terra-button@^2.1.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-button@^2.31.1, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-dialog-modal" package depends on terra-button@^2.8.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-form-validation" package depends on terra-button@^2.7.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-hookshot" package depends on terra-button@^2.3.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-popup" package depends on terra-button@^2.3.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-modal-manager" package depends on terra-button@^2.28.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-notification-dialog" package depends on terra-button@^2.5.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-slide-panel-manager" package depends on terra-button@^2.27.0, which differs from the hoisted terra-button@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-aggregator" package depends on terra-content-container@^2.4.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-layout" package depends on terra-content-container@^2.2.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-popup" package depends on terra-content-container@^2.2.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-utility" package depends on terra-content-container@^2.1.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-modal-manager" package depends on terra-content-container@^2.27.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-slide-panel-manager" package depends on terra-content-container@^2.26.0, which differs from the hoisted terra-content-container@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-doc-template@^1.22.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-embedded-content-consumer" package depends on terra-doc-template@^1.22.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-slide-group" package depends on terra-doc-template@^1.22.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-time-input" package depends on terra-doc-template@^1.22.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-dialog-modal" package depends on terra-doc-template@^1.5.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-notification-dialog" package depends on terra-doc-template@^1.2.0, which differs from the hoisted terra-doc-template@^1.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-form-validation" package depends on terra-form-input@^1.6.0, which differs from the hoisted terra-form-input@^1.28.0.
lerna WARN EHOIST_PKG_VERSION "terra-time-input" package depends on terra-form-input@^1.31.1, which differs from the hoisted terra-form-input@^1.28.0.
lerna WARN EHOIST_PKG_VERSION "terra-aggregator" package depends on terra-grid@^4.0.1, which differs from the hoisted terra-grid@^4.2.0.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-list@^2.31.1, which differs from the hoisted terra-list@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-infinite-list" package depends on terra-list@^2.1.0, which differs from the hoisted terra-list@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-popup" package depends on terra-list@^2.3.0, which differs from the hoisted terra-list@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-layout" package depends on terra-icon@^2.2.0, which differs from the hoisted terra-icon@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-utility" package depends on terra-icon@^2.1.0, which differs from the hoisted terra-icon@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-infinite-list" package depends on terra-icon@^2.1.0, which differs from the hoisted terra-icon@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-icon@^2.29.1, which differs from the hoisted terra-icon@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-notification-dialog" package depends on terra-icon@^2.6.0, which differs from the hoisted terra-icon@^2.0.1.
lerna WARN EHOIST_PKG_VERSION "terra-application-layout" package depends on terra-image@^2.2.0, which differs from the hoisted terra-image@^2.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-application-name" package depends on terra-image@^2.0.1, which differs from the hoisted terra-image@^2.1.0.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-form@^2.10.0, which differs from the hoisted terra-form@^2.2.0.
lerna WARN EHOIST_PKG_VERSION "terra-collapsible-menu-view" package depends on terra-mixins@^1.23.0, which differs from the hoisted terra-mixins@^1.0.0.
lerna WARN EHOIST_PKG_VERSION "terra-theme-provider" package depends on terra-mixins@^1.12.0, which differs from the hoisted terra-mixins@^1.0.0.
lerna WARN EHOIST_PKG_VERSION "terra-dialog-modal" package depends on terra-abstract-modal@^1.1.0, which differs from the hoisted terra-abstract-modal@^1.2.0.
lerna WARN EHOIST_PKG_VERSION "terra-modal-manager" package depends on terra-abstract-modal@^1.0.0, which differs from the hoisted terra-abstract-modal@^1.2.0.
lerna WARN EHOIST_PKG_VERSION "terra-infinite-list" package depends on terra-overlay@^2.1.0, which differs from the hoisted terra-overlay@^2.0.1.
lerna info Installing external dependencies
lerna info hoist Installing hoisted dependencies into root
lerna info hoist Pruning hoisted dependencies
lerna info hoist Finished pruning hoisted dependencies
lerna info hoist Finished installing in root
lerna info Symlinking packages and binaries
lerna info lifecycle postinstall
lerna info lifecycle prepublish
lerna info lifecycle prepare
lerna success Bootstrapped 26 packages
npm WARN terra-toolkit@4.20.0 requires a peer of nightwatch@^0.9.12 but none was installed.

added 2507 packages in 418.889s

```

</p>
</details>

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (StephenEsser) [inline .vscode/settings.json]: notsure

File: .npmrc [hunk 2]
```diff
@@ -1 +1 @@
-package-lock=false
```
Comment (bjankord) [inline .npmrc]: I'm not sure we can use package-lock with our version of lerna. There were some issues a while back that prevented us from using it. Need to look into this some more.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (bjankord) [inline .vscode/settings.json]: 🤔

**Comment by tbiethman** [comment]
@bjankord Agreed, this will be impossible to maintain otherwise

**Comment by emilyrohrbough** [comment]
@bjankord @tbiethman  I think this makes sense. I've made the changes to simplify the version.

**Pull request merged**